src/server/server.ts

```diff
@@ -37,6 +41,13 @@ export async function server(
   });
   server.register(fastifyCors);
   server.register(formBodyPlugin);
+  server.decorateRequest('previewUrl', '');
+  server.addHook('onRequest', (request, reply, done) => {
+    request.raw.previewUrl =
+      request.protocol + '://' + request.hostname + '/preview/{{id}}';
+    done();
+  });
```

Comment:
It seems the preview URL must be absolute, so we have to dynamically get the request’s hostname to build the URL.